javalib/src/main/scala/java/lang/Double.scala

```diff
@@ -86,19 +88,28 @@ object Double {
 
   @inline def valueOf(s: String): Double = valueOf(parseDouble(s))
 
-  private[this] lazy val doubleStrPat = new js.RegExp(
-      "^"                   +
-      "[\\x00-\\x20]*("     + // optional whitespace
-      "[+-]?"               + // optional sign
-      "(?:NaN|Infinity|"    + // special cases
-       "(?:\\d+\\.?\\d*|"   + // literal w/  leading digit
-        "\\.\\d+)"          + // literal w/o leading digit
-       "(?:[eE][+-]?\\d+)?" + // optional exponent
-      ")[fFdD]?"            + // optional float / double specifier (ignored)
-      ")[\\x00-\\x20]*"     + // optional whitespace
-      "$")
-
-  private[this] lazy val doubleStrHexPat = new js.RegExp(
+  private[this] lazy val doubleStrPatStr =
```

Comment:
I think there is no reason left to extract the strings in separate vals. It seems we can revert to the old structure, only with `RegExpImpl.impl.compile` instead of `new js.RegExp`.